app/src/main/java/com/example/mytestproject/ui/searchCity/SearchCityAdapter.kt

```diff
@@ -0,0 +1,66 @@
+package com.example.mytestproject.ui.searchCity
+
+import android.util.Log
+import android.view.LayoutInflater
+import android.view.ViewGroup
+import androidx.databinding.DataBindingUtil
+import androidx.recyclerview.widget.RecyclerView
+import com.example.domain.models.CityModel
+import com.example.mytestproject.R
+import com.example.mytestproject.databinding.CellCityBinding
+
+class SearchCityAdapter: RecyclerView.Adapter<SearchCityAdapter.SearchCityViewHolder>() {
+
+    private val cityModelList = mutableListOf<CityModel>()
+
+    private var onclickListenerCityModel: OnClickListenerCityModel? = null
+
+    fun attachCityModel(onclickListener: OnClickListenerCityModel) {
+        onclickListenerCityModel = onclickListener
+    }
+
+    fun setData(newCityList: List<CityModel>) {
+        cityModelList.clear()
+        cityModelList.addAll(newCityList)
+        notifyDataSetChanged()
+    }
+
+    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): SearchCityViewHolder {
+        val cellCityModel: CellCityBinding = DataBindingUtil.inflate(
+            LayoutInflater.from(parent.context),
+            R.layout.cell_city,
+            parent,
+            false
+        )
+        return SearchCityViewHolder(cellCityModel, onclickListenerCityModel)
+    }
+
+    override fun getItemCount(): Int {
+        return cityModelList.size
+    }
+
+    override fun onBindViewHolder(holder: SearchCityViewHolder, position: Int) {
+        holder.bind(cityModelList[position])
+        holder.clickItem(cityModelList[position])
+    }
+
+    class SearchCityViewHolder(cityModelItem: CellCityBinding,private val  onclickListener: OnClickListenerCityModel?) :
+        RecyclerView.ViewHolder(cityModelItem.root) {
+
+        private val cellCityModel = cityModelItem
+
+        fun bind(cityModel: CityModel) {
+            cellCityModel.cityModel = cityModel
+        }
+
+        fun clickItem(cityModel: CityModel) {
+            itemView.setOnClickListener {
+               onclickListener?.getCityModel(cityModel)
+            }
+        }
+    }
+
+    interface OnClickListenerCityModel {
+        fun getCityModel(cityModel: CityModel)
```

Comment:
Let's simplify it. Adapter shouldn't return the whole object. Only ID of clicked item. This ID will be used in presenter/viewModel to do some work.

Reply:
Fixed